src/platform/packages/shared/kbn-esql-ast/src/commands_registry/commands/rerank/autocomplete.ts

```diff
@@ -19,18 +39,253 @@ export async function autocomplete(
+        // check if fragment is completed
+        (fragment) => !!columnExists(fragment, context),
+        // incomplete: get available fields suggestions
+        (_: string, rangeToReplace?: { start: number; end: number }) => {
+          return fieldSuggestions.map((suggestion) => ({
+            ...suggestion,
+            text: `${suggestion.text}`,
+            rangeToReplace,
+            command: TRIGGER_SUGGESTION_COMMAND,
+          }));
+        },
+        // complete: get next actions suggestions for completed field
+        (fragment: string, rangeToReplace: { start: number; end: number }) => {
+          const suggestions = buildNextActions({ includeAssign: true, withSpaces: true });
+
+          return suggestions.map((suggestion) => ({
+            ...suggestion,
+            filterText: fragment,
+            text: fragment + suggestion.text,
+            rangeToReplace,
+            command: TRIGGER_SUGGESTION_COMMAND,
+          }));
+        }
+      );
+    }
+
+    case 'boolean_expression': {
+      if (expressionRoot) {
+        const expressionType = getExpressionType(
+          expressionRoot,
+          context?.fields,
+          context?.userDefinedColumns
+        );
+
+        if (expressionType === 'boolean' && isExpressionComplete(expressionType, innerText)) {
+          return buildNextActions({ includeBinaryOperators: true });
+        }
+      }
+
+      return await suggestForExpression({
+        innerText,
+        getColumnsByType: callbacks.getByType,
+        expressionRoot,
+        location: Location.WHERE,
+        preferredExpressionType: 'boolean',
+        context,
+        hasMinimumLicenseRequired: callbacks?.hasMinimumLicenseRequired,
+        activeProduct: context?.activeProduct,
+      });
+    }
+
+    default:
+      return [];
+  }
+}
+
+/**
+ * Handles suggestions within WITH map expressions
+ */
+function handleMapExpression(
+  innerText: string,
+  endpoints?: InferenceEndpointAutocompleteItem[]
+): ISuggestionItem[] {
+  const availableParameters = {
+    inference_id: endpoints?.map(createInferenceEndpointToCompletionItem) || [],
+  };
+
+  return getCommandMapExpressionSuggestions(innerText, availableParameters);
+}
 
-  // TODO: Add more specific autocomplete logic for RERANK command
-  const fieldSuggestions = await callbacks.getByType('any');
-  suggestions.push(...fieldSuggestions);
+// ============================================================================
+// Suggestion Builders and Formatters
+// ============================================================================
 
-  // Add the pipe completion item when appropriate
-  suggestions.push(pipeCompleteItem);
+function createBasicConstants(): ISuggestionItem[] {
+  return [
+    {
+      label: 'Your search query',
+      text: '"${0:Your search query.}"',
+      asSnippet: true,
+      kind: 'Constant',
+      sortText: '1',
+      detail: '',
+    },
+  ];
+}
+
+function createInferenceEndpointToCompletionItem(
```

Comment:
maybe this should be shared with COMPLETION

Reply:
It should 👍 Let's move it on a shared place and reuse. Here `src/platform/packages/shared/kbn-esql-ast/src/definitions/utils/autocomplete`